arangod/Utils/SupportInfoBuilder.cpp

```diff
@@ -0,0 +1,762 @@
+    }
+    result.openObject();
+    result.add("n_doc_colls", VPackValue(dbInfo.numDocColls));
+    result.add("n_graph_colls", VPackValue(dbInfo.numGraphColls));
+    result.add("n_smart_colls", VPackValue(dbInfo.numSmartColls));
+    result.add("n_disjoint_graphs", VPackValue(dbInfo.numDisjointGraphs));
+
+    for (auto const dbIt : velocypack::ObjectIterator(dbInfo.builder.slice())) {
+      std::string_view key = dbIt.key.stringView();
+      auto const value = dbIt.value;
+      if (key == "colls") {
+        result.add("colls", VPackValue(VPackValueType::Array));
+        for (auto const collIt : velocypack::ArrayIterator(value)) {
+          result.openObject();
+          for (auto const collIt2 : velocypack::ObjectIterator(collIt)) {
+            std::string_view key2 = collIt2.key.stringView();
+            auto const value2 = collIt2.value;
+            if (key2 == "n_docs") {
+              auto foundIt =
+                  collNumDocs.find(collIt.get("plan_id").getNumber<size_t>());
+              if (foundIt != collNumDocs.end()) {
+                result.add(key2, VPackValue(foundIt->second));
+              }
+            } else if (key2 != "name") {
+              result.add(key2, value2);
+            }
+          }
+          result.close();
+        }
+        result.close();
+
+      } else {
+        result.add(key, value);
+      }
+    }
+    result.close();
+  }
+}
+
+// if it's a telemetrics request, keys can only have lowercase letters and
+// underscores
+void SupportInfoBuilder::normalizeKeyForTelemetrics(std::string& key) {
+  key = arangodb::basics::StringUtils::tolower(key.data());
+  key = absl::StrReplaceAll(std::move(key), {{".", "_"}, {"-", "_"}});
+}
+
+void SupportInfoBuilder::buildInfoMessage(VPackBuilder& result,
+                                          std::string const& dbName,
+                                          ArangodServer& server, bool isLocal,
+                                          bool isTelemetricsReq) {
+  bool isSingleServer = ServerState::instance()->isSingleServer();
+  auto const serverId = ServerIdFeature::getId().id();
+  // used for all types of responses
+  VPackBuilder hostInfo;
+
+  buildHostInfo(hostInfo, server, isTelemetricsReq);
+
+  std::string timeString;
+  LogTimeFormats::writeTime(timeString,
+                            LogTimeFormats::TimeFormat::UTCDateString,
+                            std::chrono::system_clock::now());
+
+  bool const isActiveFailover =
+      server.getFeature<ReplicationFeature>().isActiveFailoverEnabled();
+  bool fanout =
+      (ServerState::instance()->isCoordinator() || isActiveFailover) &&
+      !isLocal;
+
+  result.openObject();
+
+  if (isSingleServer && !isActiveFailover) {
+    result.add("deployment", VPackValue(VPackValueType::Object));
+    if (isTelemetricsReq) {
+      // it's single server, but we maintain the format the same as cluster
+      std::string envValue;
+      TRI_GETENV("STARTUP_MODE", envValue);
```

Comment:
Environment variables under our control should always be prefixed with `ARANGODB_`
```suggestion
      TRI_GETENV("ARANGODB_STARTUP_MODE", envValue);
```